### ISSUE TITLE
Error: QOSMaxSubmitJobPerUserLimit

### ISSUE THREAD

**Issue by meganlsmith** [marbl/canu#1945]
Hi everyone,

I am running into an error related to grid use. Specifically, my grid has a limit of 500 jobs per user at a time, and I am getting the following error: sbatch: error: QOSMaxSubmitJobPerUserLimit

I am running the following command in canu: 

/N/project/Prophysaongenomics/canu-2.1.1/bin/canu -p prophysaon -d prophysaon  useGrid=true gridOptions="-t 2-0" genomeSize=1.54g -pacbio-raw ./fastq/*.fq corMhapFilterThreshold=0.0000000002 corMhapOptions="--threshold 0.80 --num-hashes 512 --num-min-matches 3 --ordered-sketch-size 1000 --ordered-kmer-size 14 --min-olap-length 2000 --repeat-idf-scale 50" mhapMemory=60g mhapBlockSize=500 ovlMerDistinct=0.975 gridEngineArrayMaxJobs=70

I thought that adding 'gridEngineArrayMaxJobs=70' would prevent canu from submitting more than 70 jobs at a time, but it seems that canu tries to submit multiple arrays of size 70, and if the total number of jobs exceeds 500, I end up getting the same error. Is there a way to limit this, or is my best option to use useGrid=remote?

I have attached the log file; thanks in advance for any advise you can give!

[canu.out.txt](https://github.com/marbl/canu/files/6478695/canu.out.txt)

Sincerely,
Megan Smith

**Comment by brianwalenz** [comment]
There are three ways around it.  Two are tedious and one is untested.  I guess there is a fourth way - plea to your sysadmins.

The first method is to increase the job sizes so that you end up with less than 500 jobs per stage.  There are a lot of parameters that would need to be tweaked, and it's possible that the jobs won't fit in either memory or time limits.  I don't suggest this method.

Incidentally, I'm not sure why `mhapBatchSize=500` is suggested in the FAQ, but this is resulting in many jobs; the default is 3000.

The second method is to `useGrid=remote` and manually submit jobs in batches using the canu-generated `jobSubmit` scripts.

The third method attempts to automate that.  It is detailed in issue #1883 (which also has some hints for the second approach).  The basic idea is to submit the jobs as normal, ignore the error, then let Canu restart once the jobs run.  On the restart, Canu will discover which results are computed, submit jobs for the remaining, ignore the error, and repeat until all jobs complete.  There is a small code modification needed to ignore the error' though.

**Comment by skoren** [comment]
Idle

**Comment by goeckeritz** [comment]
Hi Canu Crew, 

I wanted to to bring this issue back from the dead because I am having the same error, even though I've run Canu jobs no problem, allowing it to configure the job specifications as it sees fit (I ran it two weeks ago!) So, I have a curiosity question.  I'm getting stuck in oea steps. I was wondering if Canu asks for more resources if I set correctedErrorRate to a higher value (in this case, 0.095)? I'm using PacBio Sequel II reads and canu v1.9.
